lib/versioning/VersionID.js

```diff
@@ -117,35 +105,29 @@ function generateVersionId(info) {
 }
 
 /**
- * Encrypt a versionId to add a validation layer before processing
- * bucket versioning related requests.
+ * Encode a versionId to obscure internal information contained
+ * in a version ID.
  *
- * @param {string} str - the versionId to encrypt
- * @return {string} - the encrypted versionId
+ * @param {string} str - the versionId to encode
+ * @return {string} - the encoded versionId
  */
-function encrypt(str) {
-    const cipher = crypto.createCipher(
-            VID_CRYPTO_ALGORITHM, VID_CRYPTO_PASSWORD);
-    let encrypted = cipher.update(str,
-            VID_CRYPTO_ENCODING_TXT, VID_CRYPTO_ENCODING);
-    encrypted += cipher.final(VID_CRYPTO_ENCODING);
-    return encrypted.replace(/=/g, '');
+function encode(str) {
+    return Buffer.from(str, 'utf8').toString('hex');
 }
 
 /**
- * Decrypt a versionId. May throw a decryption exception if the input string
- * and/or the password are invalid.
+ * Decode a versionId. May throw a TypeError exception if the input string is
+ * invalid hex string.
  *
- * @param {string} str - the encrypted versionId to decrypt
- * @return {string} - the decrypted versionId
+ * @param {string} str - the encoded versionId to decode
+ * @return {string} - the decoded versionId
  */
-function decrypt(str) {
-    const decipher = crypto.createDecipher(
-            VID_CRYPTO_ALGORITHM, VID_CRYPTO_PASSWORD);
-    let decrypted = decipher.update(str,
-            VID_CRYPTO_ENCODING, VID_CRYPTO_ENCODING_TXT);
-    decrypted += decipher.final(VID_CRYPTO_ENCODING_TXT);
-    return decrypted;
+function decode(str) {
+    const result = Buffer.from(str, 'hex').toString('utf8');
```

Comment:
This throws a `TypeError` not only if not a string, but also if the string does not meet some conditions as a hex string (might be the number of characters). 
Currently we do a try-except in all the cases we currently call this function; I agree it makes more sense to do a try-except here and return an error in case of exception.